controllers/v1/admin.js

```diff
@@ -123,4 +123,76 @@ module.exports = class Admin {
 			}
 		})
 	}
+
+	/**
+   * @api {post} /project/v1/admin/deleteResource/:resourceId?type=solution
+   * Deletes a resource (program/solution) after validating admin access.
+   * @apiVersion 1.0.0
+   * @apiGroup Admin
+   * @apiSampleRequest /project/v1/admin/deleteResource/683867e60f8595db9c1b6c26?type=solution
+    * @apiParamExample {json} Response:
+    {
+    "message": "Solution and associated resources deleted successfully",
+    "status": 200,
+    "result": {
+        "solutionDeletedCount": 1,
+        "projectTemplateDeletedCount": 1,
+        "certificateTemplateDeletedCount": 1,
+        "taskDeletedCount": 0,
+        "surveyCount": 0,
+        "surveySubmissionCount": 0,
+        "observationCount": 0,
+        "observationSubmissionCount": 0,
+        "projectDeletedCount": 2
+    	}
+	}
+    * @apiUse successBody
+    * @apiUse errorBody
+    */
+	/**
+	 * Deletes a resource (program/solution) after validating admin access.
+	 *
+	 * @param {Object} req - Express request object containing user details, params, and query.
+	 * @param {Object} req.params - Contains route parameters, specifically `_id` of the resource.
+	 * @param {Object} req.query - Contains query parameters, specifically `type` (program/solution).
+	 * @param {Object} req.userDetails - Contains user roles and tenant/org info.
+	 * @returns {Promise<Object>} - Returns a success or failure response from the adminHelper.
+	 * @throws {Object} - Throws an error object with status, message, and error details if validation or deletion fails.
+	 */
+	async deleteResource(req) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				let deleteResource
+				// Check if user is authenticated and has 'admin' role
+				if (
+					req.userDetails &&
```

Comment:
@MallanagoudaB move this to middleware, someone in future has similar needs can use that